packages/deploy-script-support/src/offer.js

```diff
@@ -77,13 +77,13 @@ export const makeOfferAndFindInvitationAmount = (
       const allDepositedP = Promise.all(
         Object.entries(paymentsP).map(([keyword, paymentP]) => {
           const depositInPurse = makeDepositInPurse(keyword);
-          return paymentP.then(depositInPurse);
+          return E.when(paymentP, depositInPurse);
```

Comment:
@Chris-Hibbert this is the case I ran into when reviewing durable zoe. The `paymentP` was a payment, not a promise for a payment, and so did not have a `then` method.

Reply:
At https://github.com/Agoric/agoric-sdk/actions/runs/3701090280/jobs/6270126161

```
Rejected promise returned by test. Reason:

  TypeError {
    message: 'paymentP.then is not a function',
  }

  › packages/deploy-script-support/src/offer.js:80:27
  › Array.map (<anonymous>)
  › handlePayments (packages/deploy-script-support/src/offer.js:78:35)
  › async packages/deploy-script-support/test/unitTests/test-offer.js:91:3
```